### PR TITLE
handle hr error conditions

### PR DESCRIPTION
**Pull request by orilevari**
**Description**: SDL bug fixes for warning C6387

### CODE REVIEW

File: winml/test/scenario/cppwinrt/CustomOperatorProvider.h [hunk 1]
```diff
@@ -26,6 +26,9 @@ struct CustomOperatorProvider :
 #elif WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_PC_APP)
         m_library = LoadPackagedLibrary(winml_dll_name.c_str(), 0 /*Reserved*/);
 #endif
+        if (!m_library) {
+          __fastfail(0);
```
Comment (ryanlai2) [inline winml/test/scenario/cppwinrt/CustomOperatorProvider.h:30]: Should there be a more meaningful return code for __fastfail ?

File: winml/test/api/raw/winml_microsoft.h [hunk 2]
```diff
@@ -173,6 +173,11 @@ inline HRESULT GetActivationFactory(
 {
     // Fallback to OS binary if the redistributable is not present!
     auto library = LoadLibraryExW(MachineLearningDll, nullptr, 0);
+    if (library == nullptr)
+    {
+      auto hr = HRESULT_FROM_WIN32(GetLastError());
```
Comment (martinb35) [inline winml/test/api/raw/winml_microsoft.h:178]: can you just "return HRESULT_FROM_WIN32(GetLastError();"?

**Review by martinb35 — DISMISSED** [review]

File: winml/test/scenario/cppwinrt/CustomOperatorProvider.h (hunk 1 above)
Comment (orilevari) [reply winml/test/scenario/cppwinrt/CustomOperatorProvider.h]: Used WINML_EXPECT_TRUE instead

**Review by martinb35 — APPROVED** [review]

**Pull request merged**